src/libstd/process.rs

```diff
@@ -1418,7 +1418,8 @@ mod tests {
 
         for (ref k, ref v) in env::vars() {
             // don't check android RANDOM variables
-            if cfg!(target_os = "android") && *k == "RANDOM" {
+            if cfg!(target_os = "android") &&
+               (*k == "RANDOM" || k.contains("-")) {
```

Comment:
This feels like it's including lots of environment variables from android systems, no? I'd definitely want a comment here..